test/lib/study_cleanup_tools_test.rb

```diff
@@ -0,0 +1,105 @@
+require 'test_helper'
+
+# test all validations to ensure security is being enforced that will block accidental deletes of user studies/workspaces
+# does not directly test cleanup methods as these would destroy seed data needed for other tests
+class StudyCleanupToolsTest < ActiveSupport::TestCase
+
+  include Minitest::Hooks
+  include SelfCleaningSuite
+  include TestInstrumentor
+
+  before(:all) do
+    @user = FactoryBot.create(:user, test_array: @@users_to_clean)
+    @basic_study = FactoryBot.create(:study, name_prefix: 'Cleanup Security Checks', test_array: @@studies_to_clean)
+  end
+
+  teardown do
+    Rails.env = "test" # reset this value to prevent other env checks failing later as this will persist across all runs
+  end
+
+  test 'should validate hostname' do
+    assert StudyCleanupTools.permit_hostname?
```

Comment:
it would be really nice to have a negative test here.  Can you stub the Socket.gethostname method?

Reply:
Great point.

Reply:
32c1bc3